unity-client/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Texture/AssetPromise_Texture.cs

```diff
@@ -130,20 +126,30 @@ protected override bool AddToLibrary()
             return true;
         }
 
-        string ConstructId(string textureUrl, TextureWrapMode textureWrapMode, FilterMode textureFilterMode)
-        {
-            return ((int)textureWrapMode).ToString() + ((int)textureFilterMode).ToString() + textureUrl;
-        }
+        string ConstructId(string textureUrl, TextureWrapMode textureWrapMode, FilterMode textureFilterMode) { return ((int)textureWrapMode).ToString() + ((int)textureFilterMode).ToString() + textureUrl; }
 
         public override object GetId()
         {
             // We only use the id-with-settings when storing/reading from the library
             return idWithDefaultTexSettings;
         }
 
-        public bool UsesDefaultWrapAndFilterMode()
+        public bool UsesDefaultWrapAndFilterMode() { return wrapMode == DEFAULT_WRAP_MODE && filterMode == DEFAULT_FILTER_MODE; }
+
+        static byte[] questionMarkPNG = { 137, 80, 78, 71, 13, 10, 26, 10, 0, 0, 0, 13, 73, 72, 68, 82, 0, 0, 0, 8, 0, 0, 0, 8, 8, 2, 0, 0, 0, 75, 109, 41, 220, 0, 0, 0, 65, 73, 68, 65, 84, 8, 29, 85, 142, 81, 10, 0, 48, 8, 66, 107, 236, 254, 87, 110, 106, 35, 172, 143, 74, 243, 65, 89, 85, 129, 202, 100, 239, 146, 115, 184, 183, 11, 109, 33, 29, 126, 114, 141, 75, 213, 65, 44, 131, 70, 24, 97, 46, 50, 34, 72, 25, 39, 181, 9, 251, 205, 14, 10, 78, 123, 43, 35, 17, 17, 228, 109, 164, 219, 0, 0, 0, 0, 73, 69, 78, 68, 174, 66, 96, 130, };
+        internal bool IsQuestionMarkPNG(Texture tex)
         {
-            return wrapMode == DEFAULT_WRAP_MODE && filterMode == DEFAULT_FILTER_MODE;
+            if (!tex)
+                return true;
+
+            if (tex.width * tex.height != questionMarkPNG.Length )
```

Comment:
Should this consider the 4 channels?
`tex.width * tex.height * 4 != questionMarkPNG.Length`

Reply:
My calculations were completly off, to be honest not even taking into account the 4 channels would work.

Somehow the 8x8 image ends up in an byte array of 122 entries. Not sure what the PNG encoding is doing in the background.

I'm going to had a hardcoded check of the texture's width and height. In any case, if Unity decides to change that bogus image, the test will fail, so not the end of the world.